packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -158,6 +157,21 @@ class AssembleCommand extends FlutterCommand {
     return const <CustomDimensions, String>{};
   }
 
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async {
+    final String platform = environment.defines[kTargetPlatform];
+    if (platform == null) {
+      return super.requiredArtifacts;
+    }
+
+    final TargetPlatform targetPlatform = getTargetPlatformForName(platform);
+    final DevelopmentArtifact artifact = artifactFromTargetPlatform(targetPlatform);
+    if (artifact != null) {
```

Comment:
actually, this includes universal right?

Reply:
(I think the default required artifacts is empty)

Reply:
oh wait, nvm it does

Reply:
Yeah universal doesn't need to be explicitly returned, it always gets downloaded.

Thanks @christopherfujino for the breadcrumb #45267
https://github.com/flutter/flutter/blob/a7f7687a2e207cfc87146fb27ab7ab507228cf63/packages/flutter_tools/lib/src/cache.dart#L78-L80